### PR TITLE
Correct HDT heading with offset and direction and publish as IMU message for robot_localization

### PR DESCRIPTION
**Pull request by rcp1-beg**
Similar to [#132](https://github.com/ros-drivers/nmea_navsat_driver/pull/132) of the original repo.

### CODE REVIEW

File: launch/nmea_driver_testing.launch [hunk 1]
```diff
@@ -15,6 +15,8 @@
   <arg name="driver_rate"                 default="100" />
 
   <!-- Heading settings -->
+  <arg name="heading_as_imu"              default="False" />
```
Comment (ggaessler) [inline launch/nmea_driver_testing.launch]: The name might indicate that is enables/disabled the IMU messages on all heading messages, so also the vendor specific ones, which isn't the case. Please either use a clearer name, indicating that this only applies to NMEA HDT messages or use it as switch for all heading IMU messages.

**Review by ggaessler — CHANGES_REQUESTED** [review]
Sorry for the late reaction and thanks for the PR. Looks good to me except the argument naming, see the corresponding comment.

File: launch/nmea_driver_testing.launch (hunk 1 above)
Comment (rcp1-beg) [reply launch/nmea_driver_testing.launch]: Clarified by renaming the parameter to hdt_heading_as_imu.

**Review by ggaessler — APPROVED** [review]

**Comment by ggaessler** [comment]
Thanks for the PR and the quick renaming! Looks good to me. However the CI pipeline was a bit outdated and didn't manage to run the checks properly due to that. I just fixed the pipeline in `master`. Could you please pull the changes in, so that the checks don't fail anymore? Will merge as soon as they work again.

**Comment by rcp1-beg** [comment]
Rebased the branch in my fork on your master.

**Comment by ggaessler** [comment]
> Rebased the branch in my fork on your master.

Thanks a lot!

**Pull request merged**